SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentFactory.cpp

```diff
@@ -316,6 +316,17 @@ void ComponentFactory::AddProperty(Schema_Object* Object, Schema_FieldId FieldId
 			AddProperty(Object, FieldId, ArrayProperty->Inner, ArrayHelper.GetRawPtr(i), ClearedIds);
 		}
 
+		if (ArrayHelper.Num() > 0)
```

Comment:
Always forget, but wouldn't we want to updateInterest if the array goes from e.g. 2->0?
Or is this function used differently?

Reply:
I think you are correct - I believe the interest won't be lost on array if we clear it. I'll add a test and validate.